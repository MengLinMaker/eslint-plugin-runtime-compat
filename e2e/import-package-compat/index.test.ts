import { execSync } from 'node:child_process'
import { describe, expect, it } from 'vitest'

describe('Multi file import linting test', () => {
  it('should detect [Cache] incompatability', () => {
    try {
      execSync('pnpm test:lint', {
        cwd: import.meta.dirname,
        encoding: 'utf8',
      })
    } catch (e) {
      // @ts-expect-error error is not well defined
      expect(e.stdout).toMatchSnapshot()
    }
  })
})
