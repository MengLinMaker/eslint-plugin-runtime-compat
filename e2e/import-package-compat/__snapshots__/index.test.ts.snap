// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Multi file import linting test > should detect [Cache] incompatability 1`] = `
"
> @ test:lint /Users/menglinmaker/Documents/2-Engineering/OpenSource/eslint-runtime-compat/e2e/import-test
> eslint


/Users/menglinmaker/Documents/2-Engineering/OpenSource/eslint-runtime-compat/e2e/import-test/index.ts
  6:5  error  class - [Cache] - Unsupported standard API for [bun,edge-light,fastly,node]
Docs - https://developer.mozilla.org/docs/Web/API/Cache  runtime-compat/runtime-compat

✖ 1 problem (1 error, 0 warnings)

 ELIFECYCLE  Command failed with exit code 1.
"
`;
